### PR TITLE
[WIP] Support Angular Control Flow Syntax

### PR DESCRIPTION
**Pull request by sosukesuzuki**
# This is still a WIP.

## Description

**TODO:**

- [x] Support span in transformation
- [ ] Format embedded block parameters
- [ ] More tests

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-html/preprocess/block/transform-block.js [hunk 1]
```diff
@@ -7,22 +7,24 @@ import { transformDeferredBlock } from "./transform-defer-block.js";
...
-          transformDeferredBlock(node);
-          break;
-        case "for":
-          transformForLoopBlock(node);
-          break;
+function transformControlFlowBlockNode(ast, options) {
+  if (options.parser === "angular") {
+    ast.walk((node) => {
+      if (node.type === "block") {
+        switch (node.name) {
```
Comment (eneajaho) [inline src/language-html/preprocess/block/transform-block.js:14]: what about switch block?

Comment (sosukesuzuki) [reply src/language-html/preprocess/block/transform-block.js:14]: Switch block can be printed as the original AST structure, so there is no need to transform it.

**Comment by kbrilla** [comment]
Found small issue with formating of nested `@else if`/`@else` blocks in `@else if`/`@else` blocks. In `@if` blocks they format properly

image
[playground](https://deploy-preview-15532--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEABAlgMwAQAoYBOArnAJTbDYA6U22GO+xZFNAvvXADYDOc2WPJgCGvMpQ7tOYgYxFjSE6lBoM8hEuWAdU3PrKGi+itU02soOvf0G55xpdg4gANCAgAHGOmg9koYQICCAB3AAVAhD8UURDhAE8-NwAjAmEwAGs4GABlYQBbOAAZdCg4ZHs4FLTM7JyPdNKAc2QNKpA+fPRW5jceZq44AEUiCHgKo3aAKx4ADxyB4dHxpEq3AEdluDDgj2iQYR4AWjK4ABNz1xBCYXQuZoBhCHz84WQDri4r-qgmwYBBGCEdDJIjwMJwAglMoTMRuAAWMHyXAA6vD0PAeA0wHAclEMegAG4Y+LvMA8JIgQkkACSUAusByYAI6C8-3pORg8UGsL4bg8wT4KLSHneArgfAIhPKblKkpgO2ETVevPaDQIkvewl+RC4gSuAtKMBR6DOMHhyAAHAAGNwEOCbdD2xXKt6rSZuGDCZIms0WpAAJjcRD4ABVvdE1iA4PlkucLmcitqmkQlXAAGIQAivIHNLVgiAgNhsIA)

**Comment by kbrilla** [comment]
also `@if` after `@if` format ok but `@if` after `@else if` does not
image

[playground](https://deploy-preview-15532--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEABAlgMwAQAoYBOArnAJTbDYA6UAvtVBjvsWRQ7TTU3oSeZRr1UcADYBnONix5MAQwltgnRjJb92QkABoQEAA4x00cclByCBCAHcAChYSmUC63ICep3QCMCcsAGs4GABlOQBbOAAZdCg4ZHlFb18AoOD9PxiAc2Q+OF1JMPQc1nys0TgARSIIeHiFSV0AK3EAD2CyyurapASGkABHLrhbK30nEDlxAFpYuAATeZ0QQjl0USyAYQgwsLlkCdFRJfEOgEEYQnQvInhbOAJo2LrEkAALGDDRAHVX9HhxdJgODBRx-dAANz+bn2YHEnhA4JIAEkoAtYMEwAR0IZTqjgjA3OVnn19FZJF9fPp9qS4JICOC4roYnSYCM5JldsS8iB0gQ6fs5FBMkRRBYlqSYjAvug5jBXsgABwABl0BDgg3QarZHL2PXq3Jgci80tl8qQACZdERJAAVI1OXrcuBhLzzBZzSKC4XsuAAMQgBF2FyyApuEBAtFoQA)

**Comment by NateRadebaugh** [comment]
I was testing the playground and found another case that doesn't format quite right:
image

**Comment by sheikalthaf** [comment]
@sosukesuzuki This is really blocker for angular v17 update. Any plan to merge this PR anytime soon?

**Comment by kbrilla** [comment]
I would nit say it’s blocker for v17, You can still use it without new
control flow syntax, and wait  for it till then. Angular 17 have much more
to offer than just new CFS

W dniu śr., 8.11.2023 o 19:18 Sheik Althaf ***@***.***>
napisał(a):

**Comment by rdlabo** [comment]
This is broken case:

```html
<ion-text>{{ areaState.users | number }}人@if (areaState.users === 10000) { 以上 }</ion-text>
```

↓

```html
<ion-text
  >{{ areaState.users | number }}人<if@if (areaState.users === 10000) {
    以上
  }
>
```

```html
<ion-button fill="clear" size="large"
        >@if (!isChangedPhoto) {
        <ion-icon name="image-outline" slot="icon-only"></ion-icon>
        }</ion-button
      >
```

↓

```html
<ion-button fill="clear" size="large"
  @if (!isChangedPhoto) {
    <ion-icon name="image-outline" slot="icon-only"></ion-icon>
  }
></ion-button>
```

**Comment by sosukesuzuki** [comment]
Suppressed by https://github.com/prettier/prettier/pull/15606

**Pull request closed without merging**